packages/tx/src/eip1559Transaction.ts

```diff
@@ -197,11 +197,24 @@ export default class FeeMarketEIP1559Transaction extends BaseTransaction<FeeMark
     // Verify the access list format.
     AccessLists.verifyAccessList(this.accessList)
 
-    this.maxFeePerGas = new BN(toBuffer(maxFeePerGas === '' ? '0x' : maxFeePerGas))
-    this.maxPriorityFeePerGas = new BN(
-      toBuffer(maxPriorityFeePerGas === '' ? '0x' : maxPriorityFeePerGas)
+    const maxFeePerGasB = toBuffer(maxFeePerGas === '' ? '0x' : maxFeePerGas)
+    const maxPriorityFeePerGasB = toBuffer(
+      maxPriorityFeePerGas === '' ? '0x' : maxPriorityFeePerGas
     )
 
+    if (maxFeePerGasB.length > 0 && maxFeePerGasB[0] === 0x00) {
+      // RLP encoded integer values with leading zeroes are invalid
+      throw new Error('maxFeePerGas cannot have leading zeroes')
+    }
+
+    if (maxPriorityFeePerGasB.length > 0 && maxPriorityFeePerGasB[0] === 0x00) {
+      // RLP encoded integer values with leading zeroes are invalid
+      throw new Error('maxPriorityFeePerGas cannot have leading zeroes')
+    }
```

Comment:
Would be good if good if these checks on the extra fields can only go into this helper function in `util`, maybe this can be called with something like this.

```javscript
util.leadingZeroCheck([
  [nonceB, 'nonce' ], // Second string is for the error message
  [ maxFeePerGasB, 'maxFeePerGas' ],
])
```

Something like this.